lib/shared/bucketing/__tests__/segmentation.test.ts

```diff
@@ -1091,6 +1091,90 @@ describe('SegmentationManager Unit Test', () => {
         })
+        it('should return true if starts with filter and value starts with', () => {
+            const filter = {
+                type: 'user',
+                comparator: 'startWith',
+                values: ['testuser@'],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                true,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
+        it('should return true if ends with filter and value ends with', () => {
+            const filter = {
+                type: 'user',
+                comparator: 'endWith',
+                values: ['devcycle.com'],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                true,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
+        it('should return true if not starts with filter and value does not start with', () => {
+            const filter = {
+                type: 'user',
+                comparator: '!startWith',
+                values: ['user@'],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                true,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
+        it('should return true if not ends with filter and value does not end with', () => {
+            const filter = {
+                type: 'user',
+                comparator: '!endWith',
+                values: ['devcycle.io'],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                true,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
+        it('should return false if ends with filter with empty string', () => {
+            const filter = {
+                type: 'user',
+                comparator: 'endWith',
+                values: [''],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                false,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
+        it('should return false if ends with filter with empty string', () => {
+            const filter = {
+                type: 'user',
+                comparator: 'startWith',
+                values: [''],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                false,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
```

Comment:
Can we add this for all the new filters and contains?

Reply:
do you mean add it to `!startWith` `!endWith` ?

Reply:
yep - and the contains operator - just so we have a source of truth with regard to what we expect

Reply:
added, do you want to make sure its expecting the correct values?

Reply:
yeah added the comparators